core/src/main/java/io/grpc/AbstractChannelBuilder.java

```diff
@@ -87,6 +90,18 @@ public final BuilderT executor(ExecutorService executor) {
   }
 
   /**
+   * Provides a custom {@code User-Agent} for the application.
+   *
+   * <p>It's an optional parameter. If not provided, a default grpc-specific user agent will be
```

Comment:
In both cases a grpc-specific user agent is used.

"If provided, the provided user agent will be prepended to grpc's own user agent" or similar?